File: src/main/java/com/wxl/dyttcrawler/properties/EsStoreProperties.java
package com.wxl.dyttcrawler.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.time.Duration;

/**
 * Create by wuxingle on 2020/5/17
 * es相关配置
 */
@Data
@ConfigurationProperties(prefix = "crawler.store.es")
public class EsStoreProperties {

    private PoolProperties pool = new PoolProperties();

    @Data
    public static class PoolProperties {

        private int maxThreads = 10;
    }
}

File: src/main/kotlin/com/wxl/dyttcrawler/properties/EsStoreProperties.kt
package com.wxl.dyttcrawler.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.NestedConfigurationProperty

/**
 * Create by wuxingle on 2021/10/11
 * es相关配置
 */
@ConfigurationProperties(prefix = "crawler.store.es")
data class EsStoreProperties(

    @NestedConfigurationProperty
    var pool: EsPoolProperties = EsPoolProperties()
)

data class EsPoolProperties(

    var maxThreads: Int = 5
)

